src/index.c

```diff
@@ -899,30 +905,17 @@ static void index_entry_adjust_namemask(
 		entry->flags |= GIT_INDEX_ENTRY_NAMEMASK;
 }
 
-/* When `from_workdir` is true, we will validate the paths to avoid placing
- * paths that are invalid for the working directory on the current filesystem
- * (eg, on Windows, we will disallow `GIT~1`, `AUX`, `COM1`, etc).  This
- * function will *always* prevent `.git` and directory traversal `../` from
- * being added to the index.
- */
 static int index_entry_create(
 	git_index_entry **out,
 	git_repository *repo,
 	const char *path,
 	struct stat *st,
-	bool from_workdir)
+	unsigned int path_valid_flags)
 {
 	size_t pathlen = strlen(path), alloclen;
 	struct entry_internal *entry;
-	unsigned int path_valid_flags = GIT_PATH_REJECT_INDEX_DEFAULTS;
```

Comment:
There is a lot of churn here due to my first attempt to implement `GIT_OPT_DISABLE_INDEX_FILEPATH_VALIDATION`. My second attempt has introduced `GIT_PATH_REJECT_NOTHING` and made this churn unnecessary, but I never cleaned it up.

Note that this is unrelated to sharding.